README.md

```diff
@@ -1,26 +1,28 @@
+# Band Oracle Contract
+A smart contract for enabling the Band Protocol relay system to keep crypto and fiat
+currency prices updated #onFlow
 
-## Repository settings and configuration
-- [ ]  Repository info
-    - [ ]  Add repo description
-    - [ ]  Update website to https://onflow.org
-    - [ ]  Add relevant repository topics (i.e. `blockchain` `onflow`, etc)
-    - [ ]  Check issue labels on `.github/labels.yml` and do any commit to main to get them synced
-- [ ]  Define merge workflow (create new branch protection rule)
-    - [ ]  `main` branch rule:
-        - [ ]  **Require pull request reviews before merging (2 approving reviews)**
-            - [ ]  **Require review from Code Owners**
-        - [ ]  **Require status checks to pass before merging**
-            - [ ]  **Require branches to be up to date before merging**
-        - [ ]  **Require linear history**
-        - [ ]   **Restrict who can push to matching branches**
-            - [ ]  Choose `onflow/flow` team
-- [ ]  Add necessary team members, adjust access levels
-    - [ ]  `onflow/flow-admin` ⇒ Admin access
-    - [ ]  `onflow/flow-engineering ` ⇒ Write access
+## How it works?
+The contract keeps a record of symbols and it's related market data 
+that can be updated only by authorized parties and that can be queried
+via script by any user of the Flow blockchain.
 
-## Dealing with private keys
-Get used to define your private keys on the `.env` file rather than straight on the `flow.json` file.
-Uncomment the `.env` entry on the `.gitignore` file to make sure you don't share any private keys by mistake.
\ No newline at end of file
+### Storing the data
+Market for each symbol is stored on a dictionary as a contract level field `access(contract) let symbolsRefData: {String: RefData}`. The `RefData` structs stores the following information: 
+```cadence
+    pub struct RefData {
+        // USD-rate, multiplied by 1e9.
+        pub var rate: UInt64
+        // UNIX epoch when data is last resolved. 
+        pub var timestamp: UInt64
+        // BandChain request identifier for this data.
+        pub var requestID: UInt64
+    }
+```
+This struct allows users to know when was the latest rate received and the oracle to discard updates from any relayer that are not up to date. Keep in mind that all data is stored by its USD-rate, and calculated in any other currency based on that.
+
+### Updating the data
+The account where the contract is deployed will be granted with a `OracleAdmin` resource. This resource could be use to create `DataUpdater` resources and publishing a `{&RelayUpdate}` capability to them. An authorized account can claim said capability at the time of creating a `Relay` resource. This resource will grant the ability to call the `updateData` method that will call contract function `access(contract) fun updateRefData (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64)` that will finally update the `symbolsRefData` dictionary.
```

Comment:
```suggestion
The account where the contract is deployed will be granted an `OracleAdmin` resource. This resource can create `DataUpdater` resources and publish a `{&RelayUpdate}` capability to them. An authorized account can claim said capability at the time of creating a `Relay` resource. This resource will grant the ability to call the `updateData` method that will call contract function `access(contract) fun updateRefData (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64)` that will finally update the `symbolsRefData` dictionary.
```

Reply:
Actually writing the transaction I realise that the capability will actually be published on a tx, or we will need to fall into the anti-pattern an `AuthAccount` resource as a parameter to a function (which I'm actually not horrified about). Will commit this and update it depending on how the code finally looks